microxrce_agent.cpp

```diff
@@ -200,17 +299,44 @@ int main(int argc, char** argv)
                 cfmakeraw(&attr);
                 tcflush(fd, TCIOFLUSH);
 
+                /*
+                 * Parse Baudrate.
+                 */
+                speed_t baudrate = B115200;
+                if ((cl_counter <= cl.size()) && ("--baudrate" == cl[cl_counter -1]))
+                {
+                    cl_counter++;
+                    if (cl_counter <= cl.size())
+                    {
+                        baudrate = getBaudRate(cl[cl_counter - 1].c_str());
+                        cl_counter++;
+                    }
+                }
+
                 /* Setting BAUD RATE. */
-                speed_t baudrate = (2 == cl.size()) ? getBaudRate(cl[1].c_str()) : B115200;
-                if (0 != baudrate)
+                cfsetispeed(&attr, baudrate);
+                cfsetospeed(&attr, baudrate);
+
+                if (0 == tcsetattr(fd, TCSANOW, &attr))
                 {
-                    cfsetispeed(&attr, baudrate);
-                    cfsetospeed(&attr, baudrate);
+                    std::cout << "Device: " << dev << std::endl;
+                    server.reset(new eprosima::uxr::SerialServer(fd, 0x00));
+                }
 
-                    if (0 == tcsetattr(fd, TCSANOW, &attr))
+                /*
+                 * Parse Refs File.
+                 */
+                if ((cl_counter <= cl.size()) && ("--refs" == cl[cl_counter - 1]))
+                {
+                    cl_counter++;
+                    if (cl_counter <= cl.size())
+                    {
+                        server->load_config_file(cl[cl_counter -1]);
```

Comment:
Shall we do something with the returned bool?

Reply:
Fixed!